.github/workflows/ci.yml

```
@@ -0,0 +1,95 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+    tags: ['*']
+    paths-ignore:
+      - 'CITATION.bib'
+      - 'LICENSE.md'
+      - 'README.md'
+      - '.zenodo.json'
+      - '.github/workflows/CompatHelper.yml'
+      - '.github/workflows/SpellCheck.yml'
+      - '.github/workflows/TagBot.yml'
+      - 'docs/**'
+  pull_request:
+    paths-ignore:
+      - 'CITATION.bib'
+      - 'LICENSE.md'
+      - 'README.md'
+      - '.zenodo.json'
+      - '.github/workflows/CompatHelper.yml'
+      - '.github/workflows/SpellCheck.yml'
+      - '.github/workflows/TagBot.yml'
+      - 'docs/**'
+  workflow_dispatch:
+
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: ${{ matrix.os }} - Julia ${{ matrix.version }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.8'
+          - '1.9'
+          - '1'
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+    steps:
+      - name: Check out project
+        uses: actions/checkout@v4
+      - name: Set up Julia
+        uses: julia-actions/setup-julia@v1
+        with:
+          version: ${{ matrix.version }}
+      - run: julia -e 'using InteractiveUtils; versioninfo(verbose=true)'
+      - uses: julia-actions/cache@v1
+      - name: Apply JuliaFormatter and check format
+        run: |
+          using Pkg
+          Pkg.add("JuliaFormatter")
+
+          using JuliaFormatter
+
+          is_unchanged = format(".", verbose=true)
+          if !is_unchanged
+            error("Code not properly formatted")
+          end
+        shell: julia --color=yes {0}
```

```suggestion
      - name: Apply JuliaFormatter and check format
        if: matrix.os == 'ubuntu-latest'
        run: |
          using Pkg
          Pkg.add("JuliaFormatter")
          using JuliaFormatter

          is_unchanged = format(".", verbose=true)
          if !is_unchanged
            error("Code not properly formatted")
          end
        shell: julia --color=yes {0}
```
This is too annoying, let's just skip the format check on non-Unix platforms, shall we?

---

Done.